### ISSUE TITLE
Assert out of bounds in CmdPipelineBarrier2 after CmdBeginRendering

### ISSUE THREAD

**Issue by rikusalminen** [KhronosGroup/Vulkan-ValidationLayers#4052]
**Describe the Issue**
Assert in std::vector::operator[] when validating vkCmdPipelineBarrier2 after using CmdBeginRendering from VK_EXT_dynamic_rendering. Backtrace pasted below.

My code is not creating RenderPass objects (and uses CmdBeginRendering instead), but the validation layers inspect the RenderPass state anyway.

I assume the offending line is this (in CoreChecks::ValidateRenderPassPipelineBarriers core_validation.cpp:9326)
```
    const auto &sub_desc = rp_state->createInfo.pSubpasses[state.active_subpass];
```

**Valid Usage ID**
If applicable, please include the validation messages encountered leading up to the issue

Note that there are some unrelated validation messages in the beginning of the log below.

```
VUID-vkCmdCopyImage2-renderpass(ERROR / SPEC): msgNum: -1433820550 - Validation Error: [ VUID-vkCmdCopyImage2-renderpass ] Object 0: handle = 0x7fffe80105f0, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0xaa89a27a | vkCmdCopyImage2: It is invalid to issue this call inside an active VkRenderPass 0x0[]. The Vulkan spec states: This command must only be called outside of a render pass instance (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdCopyImage2-renderpass)
    Objects: 1
        [0] 0x7fffe80105f0, type: 6, name: NULL
VUID-vkCmdBlitImage2-renderpass(ERROR / SPEC): msgNum: 1534945699 - Validation Error: [ VUID-vkCmdBlitImage2-renderpass ] Object 0: handle = 0x7fffe80105f0, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x5b7d69a3 | vkCmdBlitImage2: It is invalid to issue this call inside an active VkRenderPass 0x0[]. The Vulkan spec states: This command must only be called outside of a render pass instance (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdBlitImage2-renderpass)
    Objects: 1
        [0] 0x7fffe80105f0, type: 6, name: NULL
VUID-vkCmdBlitImage2-renderpass(ERROR / SPEC): msgNum: 1534945699 - Validation Error: [ VUID-vkCmdBlitImage2-renderpass ] Object 0: handle = 0x7fffe80105f0, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x5b7d69a3 | vkCmdBlitImage2: It is invalid to issue this call inside an active VkRenderPass 0x0[]. The Vulkan spec states: This command must only be called outside of a render pass instance (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdBlitImage2-renderpass)
    Objects: 1
        [0] 0x7fffe80105f0, type: 6, name: NULL
/usr/include/c++/11.2.0/bits/stl_vector.h:1063: std::vector<_Tp, _Alloc>::const_reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) const [with _Tp = std::vector<unsigned int>; _Alloc = std::allocator<std::vector<unsigned int> >; std::vector<_Tp, _Alloc>::const_reference = const std::vector<unsigned int>&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```

**Environment:**
 - OS: Linux
 - GPU: Intel/Mesa
 - SDK or header version if building from repo: 1.3.204 (from arch linux packages)
 - Options enabled (synchronization, best practices, etc.): VK_EXT_dynamic_rendering and synchronization2 enabled

**Additional context**

```
        [0] 0x7fffe80105f0, type: 6, name: NULL
/usr/include/c++/11.2.0/bits/stl_vector.h:1063: std::vector<_Tp, _Alloc>::const_reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) const [with _Tp = std::vector<unsigned int>; _Alloc = std::allocator<std::vector<unsigned int> >; std::vector<_Tp, _Alloc>::const_reference = const std::vector<unsigned int>&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.

Thread 3 "" received signal SIGABRT, Aborted.
[Switching to Thread 0x7ffff515e640 (LWP 674315)]
0x00007ffff7d1d34c in __pthread_kill_implementation () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff7d1d34c in __pthread_kill_implementation () from /usr/lib/libc.so.6
#1  0x00007ffff7cd04b8 in raise () from /usr/lib/libc.so.6
#2  0x00007ffff7cba534 in abort () from /usr/lib/libc.so.6
#3  0x00007ffff5fd4dca in std::__replacement_assert (__file=<optimized out>, __line=<optimized out>, __function=<optimized out>, __condition=<optimized out>) at /usr/include/c++/11.2.0/x86_64-pc-linux-gnu/bits/c++config.h:514
#4  0x00007ffff6132caf in std::vector<std::vector<unsigned int, std::allocator<unsigned int> >, std::allocator<std::vector<unsigned int, std::allocator<unsigned int> > > >::operator[] (this=<optimized out>, __n=<optimized out>) at /usr/include/c++/11.2.0/bits/stl_vector.h:1063
#5  std::vector<std::vector<unsigned int, std::allocator<unsigned int> >, std::allocator<std::vector<unsigned int, std::allocator<unsigned int> > > >::operator[] (this=<optimized out>, this=<optimized out>, __n=<optimized out>) at /usr/include/c++/11.2.0/bits/stl_vector.h:1061
#6  CoreChecks::ValidateRenderPassPipelineBarriers (this=0x55555643da10, outer_loc=..., cb_state=0x7fffe8012fb0, dep_info=0x7ffff515d100) at /build/vulkan-validation-layers/src/Vulkan-ValidationLayers-sdk-1.3.204.1/layers/core_validation.cpp:9326
#7  0x00007ffff6137f0d in CoreChecks::ValidateCmdPipelineBarrier2 (this=0x55555643da10, commandBuffer=0x7ffff515ae90, pDependencyInfo=0x7ffff515d100, cmd_type=CMD_PIPELINEBARRIER2) at /build/vulkan-validation-layers/src/Vulkan-ValidationLayers-sdk-1.3.204.1/layers/core_validation.cpp:9723
#8  0x00007ffff603919c in vulkan_layer_chassis::CmdPipelineBarrier2 (commandBuffer=0x7fffe80105f0, pDependencyInfo=0x7ffff515d100) at /build/vulkan-validation-layers/src/Vulkan-ValidationLayers-sdk-1.3.204.1/layers/generated/chassis.cpp:4868
#9  0x0000555555699001 in ash::device::Device::cmd_pipeline_barrier2 (self=0x55555624b560, command_buffer=..., dependency_info=0x7ffff515d100) at /home//.cargo/registry/src/github.com-1ecc6299db9ec823/ash-0.37.0+1.3.209/src/device.rs:117
```

**Comment by ziga-lunarg** [comment]
I believe this should already be fixed in the SDK version 1.3.211.0, can you test and confirm? @rikusalminen

**Comment by ziga-lunarg** [comment]
Pretty sure this is already fixed so I am closing the issue. If you still see the error with the latest SDK you can reopen.
